tests/integration/models/api-key.spec.ts

```diff
@@ -124,6 +124,44 @@ describe('API Key model', function () {
 		});
 	});
 
+	describe('balena.models.apiKey.getProvisioningApiKeysByApplication', function () {
+		givenLoggedInUser(before);
+		givenAnApplication(before);
+
+		before(async function () {
+			const apiKey = await balena.models.application.generateProvisioningKey(
+				this.application.id,
+			);
+			expect(apiKey).to.be.a('string').that.has.length(32);
+
+			this.apiKey = apiKey;
+		});
+
+		it('should fail when the application does not exist', function () {
+			return expect(
+				balena.models.apiKey.getProvisioningApiKeysByApplication(
+					'nonExistentApp',
+				),
+			).to.be.rejected.then((error) => {
+				expect(error).to.have.property('code', 'BalenaApplicationNotFound');
+
+				return expect(error)
+					.to.have.property('message')
+					.that.matches(/\bunique\b/i);
```

Comment:
Well, the returned error is not a unique error, so this fails atm.
Let's just drop this expectation since we already have checked for the `BalenaApplicationNotFound` code, and it should be enough.